frontend/packages/operator-lifecycle-manager/src/utils.ts

```diff
@@ -0,0 +1,23 @@
+import * as _ from 'lodash';
+import { Annotation, ClusterServiceVersionKind } from './types';
+
+export const getInternalObjects = (annotations: Annotation): string[] => {
+  const INTERNAL = 'operators.operatorframework.io/internal-objects';
+  let internals: string = _.get(annotations, INTERNAL);
+  if (internals) {
+    internals = internals.replace(/'/g, '"');
+    return JSON.parse(internals);
+  }
+  return [];
+};
+
+export const isInternalObject = (internalObjects: string[], objectName: string): boolean =>
+  internalObjects.some((obj) => obj.includes(objectName));
+
+export const internalObjectsKinds = (
+  internalObjects: string[],
+  csv: ClusterServiceVersionKind,
+): string[] => {
+  const { owned } = csv.spec.customresourcedefinitions;
+  return owned.filter((obj) => isInternalObject(internalObjects, obj.name)).map((obj) => obj.kind);
```

Comment:
`kind` is not unique since it doesn't contain the API group, so we can't use this helper to hide items by kind

Reply:
ack. will use referenceForProvidedAPI.